lib/one_model.ex

```diff
@@ -219,32 +219,39 @@ defmodule OneModel do
         {expressions, opts} = Keyword.split(opts, ~w(select order_by preload limit)a)
 
         opts
-        |> Enum.reduce(@schema, fn {k, v}, query ->
-          where(query, [b], field(b, ^k) == ^v)
-        end)
+        |> Enum.reduce(@schema, &do_where/2)
         |> do_order(expressions[:order_by])
         |> do_preload(expressions[:preload])
         |> do_select(expressions[:select])
         |> do_limit(expressions[:limit])
       end
 
+      defp do_where({k, nil}, query), do: where(query, [b], is_nil(field(b, ^k)))
+      defp do_where({k, v}, query), do: where(query, [b], field(b, ^k) == ^v)
+
       defp do_order(query, nil), do: query
       defp do_order(query, false), do: query
       defp do_order(query, []), do: query
-      defp do_order(query, order), do: order_by(query, ^order)
+
+      defp do_order(query, order) when is_list(order) or is_atom(order),
```

Comment:
I'm not sure why the new guard is needed, but if it is, then you have added the possibly of an exception with it. Is that your intent here? Please justify.